src/server/qgsserverfeatureid.cpp

```diff
@@ -0,0 +1,115 @@
+/***************************************************************************
+                              qgsserverfeatureid.cpp
+                              -----------------------
+  begin                : May 17, 2019
+  copyright            : (C) 2019 by René-Luc DHONT
+  email                : rldhont at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsserverfeatureid.h"
+#include "qgsfeature.h"
+#include "qgsfeaturerequest.h"
+#include "qgsvectordataprovider.h"
+#include "qgsexpression.h"
+
+QString QgsServerFeatureId::getServerFid( const QgsFeature &feature, const QgsAttributeList &pkAttributes )
```

Comment:
It's not very important, but generally speaking, I prefer when getter methods are not prefixed with `get`. In this case it could be `serverFid` instead of `getServerFid`.

Reply:
Actually, this is not a getter, but rather a `convertTo` or `create` method

Reply:
>  it should rather be called `PersistantFid` than `ServerFid` ;)

Indeed :)